openpype/hosts/maya/plugins/publish/validate_arnold_scene_source_cbid.py

```diff
@@ -0,0 +1,70 @@
+import maya.cmds as cmds
+
+import pyblish.api
+from openpype.pipeline.publish import (
+    ValidateContentsOrder, PublishValidationError, RepairAction
+)
+
+
+class ValidateArnoldSceneSourceCbid(pyblish.api.InstancePlugin):
+    """Validate Arnold Scene Source Cbid.
+
+    It is required for the proxy and content nodes to share the same cbid.
+    """
+
+    order = ValidateContentsOrder
+    hosts = ["maya"]
+    families = ["ass"]
+    label = "Validate Arnold Scene Source CBID"
+    actions = [RepairAction]
+
+    @staticmethod
+    def _get_nodes_data(nodes):
+        nodes_by_name = {}
+        for node in nodes:
+            node_split = node.split("|")
+            nodes_by_name[node_split[-1].split(":")[-1]] = node
+
+        return nodes_by_name
+
+    def get_invalid_couples(self, instance):
+        content_nodes_by_name = self._get_nodes_data(
+            instance.data["setMembers"]
+        )
+        proxy_nodes_by_name = self._get_nodes_data(
+            instance.data.get("proxy", [])
+        )
+
+        invalid_couples = []
+        for content_name, content_node in content_nodes_by_name.items():
+            for proxy_name, proxy_node in proxy_nodes_by_name.items():
+                if content_name == proxy_name:
+                    content_value = cmds.getAttr(content_node + ".cbId")
+                    proxy_value = cmds.getAttr(proxy_node + ".cbId")
+                    if content_value != proxy_value:
+                        invalid_couples.append((content_node, proxy_node))
+
+        return invalid_couples
+
+    def process(self, instance):
+        # Proxy validation.
+        if not instance.data.get("proxy", []):
+            return
+
+        # Validate for proxy nodes sharing the same cbId as content nodes.
+        invalid_couples = self.get_invalid_couples(instance)
+        if invalid_couples:
+            raise PublishValidationError(
+                "Found proxy nodes with mismatching cbid:\n{}".format(
+                    invalid_couples
+                )
+            )
+
+    @classmethod
+    def repair(cls, instance):
+        for content_node, proxy_node in cls.get_invalid_couples(cls, instance):
```

Comment:
I see it now. You passed it `cls` explicitly to the function! Note that the behavior then is that `self` is NOT an instance of the class but is the actual class type itself. It doesn't introduce bugs currently because you're not doing anything with attrributes on `self` but I'd better clean this up and just make `get_invalid_couples` a classmethod so the behavior is clearer.

Some examples:

How you should call `fn`, but it doesn't work.
```python
class Test(object):
    
    def __init__(self):
        self.foobar = "a"
    
    def fn(self):
        print(self.foobar)
        
    @classmethod
    def class_fn(cls):
        cls.fn()
        
Test.class_fn()
# TypeError: fn() missing 1 required positional argument: 'self' # 
```

If I pass the class then unexpected things can happen - e.g. `__init__` was never triggered because we passed the class and not an instance of the class:
```python
class Test(object):
    
    def __init__(self):
        self.foobar = "a"
    
    def fn(self):
        print(self.foobar)
        
    @classmethod
    def class_fn(cls):
        cls.fn(cls)
        
Test.class_fn()
# AttributeError: type object 'Test' has no attribute 'foobar' # 
```

But worse it could introduce bugs like:
```python


class Test(object):
    
    value = 0
    
    def set_value(self, value):
        self.value = value
        
    @classmethod
    def class_fn(cls, value):
        cls.set_value(cls, value)
        

a = Test()
print('a.value', a.value)
# a.value 0

b = Test()
b.set_value(2)
print('b.value', b.value)
# b.value 2

c = Test()
print('c.value', c.value)
# c.value 0

# After calling this we've now actually changed the
# value on the class type and not the instance. As such
# now the default value is always changed.
Test.class_fn(3)

print('a.value', a.value)
# a.value 3
print('b.value', b.value)
# b.value 2
print('c.value', c.value)
# c.value 3

# Note how classes which never got `self.value` assigned explicitly
# are still set to the class value and thus now suddenly all got changed
# and worse, new class instances will now also default to the value.
d = Test()
print('d.value', d.value)
# d.value 3

# All because we ended up actually setting:
print("Test.value", Test.value)
```